gcloud/storage/connection.py

```diff
@@ -235,6 +235,9 @@ def api_request(self, method, path, query_params=None,
         if not 200 <= response.status < 300:
             raise make_exception(response, content)
 
+        if not isinstance(content, str):
```

Comment:
This needs a comment.

I am very wary of using `str` to intentially mean `bytes` in Py2 and `unicode` in Py3. We should instead use things like `six.binary_type` and `six.text_type`.